cmd/deploy.go

```diff
@@ -15,107 +15,395 @@
+
+			removeErr := dockerRemove(extractContainerName)
+			if removeErr != nil {
+				Error.log("dockerRemove error ", removeErr)
+			}
+
+			return errors.Errorf("Error attempting to run copy command %s on image %s", bashCmd, stackImage)
+
+		}
+		//If everything went fine, we need to set the source project directory to /tmp/...
+		configDir = extractContainerName + ":" + filepath.Join("/tmp", containerConfigDir)
+	}
+	cmdArgs = []string{"cp", configDir, "./" + configFile}
+	err = execAndWaitReturnErr(cmdName, cmdArgs, Debug)
+	if err != nil {
+		Error.log("docker cp command failed: ", err)
+
+		removeErr := dockerRemove(extractContainerName)
+		if removeErr != nil {
+			Error.log("dockerRemove error ", removeErr)
+		}
+		return errors.Errorf("docker cp command failed: %v", err)
+	}
+
+	removeErr := dockerRemove(extractContainerName)
+	if removeErr != nil {
+		Error.log("dockerRemove error ", removeErr)
+	}
+
+	yamlReader, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return errors.Errorf("Config file does not exist %s. ", configFile)
+
+		}
+		return errors.Errorf("Failed reading file %s", configFile)
+
+	}
+
+	projectName, perr := getProjectName()
+	if perr != nil {
+		return errors.Errorf("%v", perr)
+	}
+
+	port, err := getEnvVarInt("PORT")
+	if err != nil {
+		//try and get the exposed ports and use the first one
+		Warning.log("Could not detect a container port (PORT env var).")
+		portsStr, portsErr := getExposedPorts()
+		if portsErr != nil {
+			return portsErr
+		}
+		if len(portsStr) == 0 {
+			//No ports exposed
+			Warning.log("This container exposes no ports. The service will not be accessible.")
+			port = 0 //setting this to 0
+		} else {
+			portStr := portsStr[0]
+			Warning.log("Picking the first exposed port as the KNative service port. This may not be the correct port.")
+			port, err = strconv.Atoi(portStr)
+			if err != nil {
+				Warning.log("The exposed port is not a valid integer. The service will not be accessible.")
+				port = 0
+			}
+		}
+	}
+	portStr := strconv.Itoa(port)
+
+	split := strings.Split(stackImage, ":")
+	stack := split[len(split)-2]
+	split = strings.Split(stack, "/")
+	stack = split[len(split)-1]
+
+	output := bytes.Replace(yamlReader, []byte("APPSODY_PROJECT_NAME"), []byte(projectName), -1)
+	output = bytes.Replace(output, []byte("APPSODY_DOCKER_IMAGE"), []byte(projectName), -1)
+	output = bytes.Replace(output, []byte("APPSODY_STACK"), []byte(stack), -1)
+	output = bytes.Replace(output, []byte("APPSODY_PORT"), []byte(portStr), -1)
+
+	err = ioutil.WriteFile(configFile, output, 0666)
+	if err != nil {
+		return errors.Errorf("Failed to write local application configuration file: %s", err)
+	}
+
+	Info.log("Created deployent manifest: ", configFile)
+	return nil
+}
+
 func init() {
 	rootCmd.AddCommand(deployCmd)
+	deployCmd.PersistentFlags().BoolVar(&generate, "generate", false, "Only generate the deployment configuration file. Do not deploy the project.")
```

Comment:
Would `--generate-only` or `--generate-manifest` or `--generate-yaml` be more meaningful?

Reply:
Switched to use `--generate-only`